scripts/mirror.sh

```diff
@@ -0,0 +1,212 @@
+              if has("variant") then .variant = "'${variant}'" else . end
+            ' <${tag}/${f} >${tag}/${f}.tmp 2>/dev/null; then
+            mv -f ${tag}/${f}.tmp ${tag}/${f}
+            tar -C ${tag} -uf ${tag}.tar ${f}
+          fi
+        done
+        docker image load -q -i ${tag}.tar >/dev/null
+        rm -rf ${tag}*
+      }
+
+      annotate-manifest() {
+        info "... Annotate platform ${platform} manifest ${mirror}"
+        docker push ${tag} >/dev/null
+        local digest=$(docker image inspect ${tag} | jq -r '.[] | .RepoDigests[0]')
+        docker manifest create ${manifest_amend} ${mirror} ${digest} >/dev/null
+        docker manifest annotate ${mirror} ${digest} ${annotate_args} >/dev/null
+        docker image rm -f ${arch_img} ${tag} >/dev/null
+        manifest_amend='--amend'
+      }
+
+      {
+        tag-arch-img
+        set-image-platform
+        annotate-manifest
+      }
+
+    done < <(manifest-list ${image})
+    info "--- Push mirror ${mirror}"
+    docker manifest push ${mirror} >/dev/null
+  }
+
+  lock "${mirror_lock}"
+  (
+    trap 'cleanup-mirror ${mirror} ${mirror_lock}' EXIT
+    info "+++ Create mirror ${mirror}"
+    process-manifest    
+  )
+}
+
+manifest-inspect() {
+  # docker manifest inspect $1 | \
+  #     jq -r '.manifests[] | "'$1'@\(.digest) \(.platform.os) \(.platform.architecture) \(.platform.variant // "")"'
+  manifest-tool inspect --raw $1 2>/dev/null | \
+      jq -r '.[] | 
+        select((.Platform.architecture // "") != "") |
+        "'$1'@\(.Digest) \(.Platform.os) \(.Platform.architecture) \(.Platform.variant // "")"
+      '
+}
+
+manifest-template() {
+  local image=${1#docker.io/}
+  local repo=${image%:*}
+  local version=${image#$repo:}
+  info "??? Generating manifest template for ${image}"
+  for platform in $(tr ',' ' ' <<<${PLATFORMS}); do
+    read -r os arch variant <<<$(tr '/' ' ' <<<${platform})
+    local img=$(sed -e "
+      s|REPO|${repo}|g;
+      s|VERSION|${version}|g;
+      s|OS|${os}|g;
+      s|ARCH|${arch}|g;
+      s|VARIANT|${variant}|g;
+    " <<<${TEMPLATE})
+    echo "${img} ${os} ${arch} ${variant}"
+  done
+}
+
+manifest-list() {
+  manifest=$(manifest-inspect $1)
+  manifest=${manifest:-$(manifest-template $1)}
+  echo "${manifest}"
+}
+
+create-mirrors() {
+  info "+++ Create mirrors"
+  local pids=()
+  local code=0
+  wait-pid() {
+    wait $1 || code=$((code+1))
+  }
+
+  for mirror in $@; do
+    [ -z "${mirror}" ] &&  continue
+    create-mirror ${mirror} &
+    local pid=$!
+    if [[ "${PARALLEL}" = 'true' ]]; then
+      pids+=(${pid})
+    else
+      wait-pid ${pid}
+    fi
+  done
+  for pid in ${pids[@]}; do
+    wait-pid ${pid}
+  done
+
+  info "--- Done create mirrors"
+  if [ $code -ne 0 ]; then
+    info "!!! Failed $code mirrors"
+  fi
+  return ${code}
+}
+
+clear-cache() {
+  info "... Clear cache"
+  rm -rf "${HOME}/.docker/manifests/"
+  docker image prune -a -f >/dev/null
```

Comment:
wondering if that can affect the shared drone instance when we attach the script to the ci